source/dubregistry/dbcontroller.d

```diff
@@ -39,9 +39,14 @@ class DbController {
 		//
 
 		// create indices
-		m_packages.ensureIndex([tuple("name", 1)], IndexFlags.Unique);
-		m_packages.ensureIndex([tuple("stats.score", 1)]);
-		m_downloads.ensureIndex([tuple("package", 1), tuple("version", 1)]);
+		IndexModel[3] indices;
+		indices[0].options.unique = true;
+		indices[0].add("name", 1);
+		indices[1].add("stats.score", 1);
+		indices[2].add("package", 1);
+		indices[2].add("version", 1);
+		m_packages.createIndexes(indices[0 .. 2]);
+		m_downloads.createIndexes(indices[2 .. 3]);
```

Comment:
I think this is quite unreadable, you should probably create the index models here inside the method.

IIRC the API should allow for this:

```d
m_packages.createIndexes([
	IndexModel().add("name", 1).withOptions(IndexOptions.unique),
	IndexModel().add("stats.score", 1)
]);
m_downloads.createIndex(IndexModel().add("package", 1).add("version", 1));
```

since they aren't actually sharing any items, even though they are part of the same array.

Reply:
Actually the `withOptions` does not work, anything I missed ?